packages/core/src/WorkflowExecute.ts

```diff
@@ -830,11 +820,16 @@ export class WorkflowExecute {
 		let closeFunction: Promise<void> | undefined;
 
 		return new PCancelable(async (resolve, reject, onCancel) => {
```

Comment:
once this PR is merged, I'm going to get rid of `PCancelable` and use a regular promise with `AbortController` instead. 
Even `p-cancelable` [recommends](https://github.com/sindresorhus/p-cancelable/commit/5e8b26853d994c1de14232ff839945fc331d2912) to do that since 3 years.
Moving away from `PCancelable` should also make this code more readable.